nibabies/workflows/anatomical/registration.py

```diff
@@ -401,68 +408,83 @@ def init_concat_registrations_wf(
+        name='dis_std2int',
     )
-    merge_std2anat_composites = pe.Node(
-        niu.Merge(1, ravel_inputs=True),
-        name='merge_std2anat_composites',
+
+    dis_int2anat = pe.Node(
+        CompositeTransformUtil(process='disassemble', output_prefix='int2anat'),
+        name='dis_int2anat',
     )
 
+    order_anat2std = pe.Node(niu.Merge(4), name='order_anat2std', run_without_submitting=True)
+    order_std2anat = pe.Node(niu.Merge(4), name='order_std2anat', run_without_submitting=True)
+
     assemble_anat2std = pe.Node(
         CompositeTransformUtil(process='assemble', out_file='anat2std.h5'),
         name='assemble_anat2std',
     )
+    # https://github.com/ANTsX/ANTs/issues/1827
+    # Until CompositeTransformUtil accepts warps as first transform,
+    # Use SimpleITK to concatenate
     assemble_std2anat = pe.Node(
-        CompositeTransformUtil(process='assemble', out_file='std2anat.h5'),
+        niu.Function(function=_create_inverse_composite, output_names=['out_file']),
         name='assemble_std2anat',
     )
 
-    fmt_cohort = pe.MapNode(
-        niu.Function(function=_fmt_cohort, output_names=['template', 'spec']),
-        name='fmt_cohort',
-        run_without_submitting=True,
-        iterfield=['template', 'spec'],
-    )
-
     workflow.connect([
-        # Template concatenation
-        (inputnode, merge_anat2std, [('anat2std_xfm', 'in2')]),
-        (inputnode, merge_std2anat, [('std2anat_xfm', 'in2')]),
+        # Transform concatenation
+        (inputnode, dis_anat2int, [('anat2int_xfm', 'in_file')]),
+        (inputnode, dis_int2anat, [('int2anat_xfm', 'in_file')]),
         (inputnode, intermed_xfms, [('intermediate', 'intermediate')]),
         (inputnode, intermed_xfms, [('template', 'std')]),
-        (intermed_xfms, merge_anat2std, [('int2std_xfm', 'in1')]),
-        (intermed_xfms, merge_std2anat, [('std2int_xfm', 'in1')]),
-        (merge_anat2std, disassemble_anat2std, [('out', 'in_file')]),
-        (merge_std2anat, disassemble_std2anat, [('out', 'in_file')]),
-        (disassemble_anat2std, merge_anat2std_composites, [('out_transforms', 'in1')]),
-        (disassemble_std2anat, merge_std2anat_composites, [('out_transforms', 'in1')]),
-        (merge_anat2std_composites, assemble_anat2std, [('out', 'in_file')]),
-        (merge_std2anat_composites, assemble_std2anat, [('out', 'in_file')]),
+        (intermed_xfms, dis_int2std, [('int2std_xfm', 'in_file')]),
+        (intermed_xfms, dis_std2int, [('std2int_xfm', 'in_file')]),
+        (dis_anat2int, order_anat2std, [
+            ('affine_transform', 'in1'),
+            ('displacement_field', 'in2'),
+        ]),
+        (dis_int2std, order_anat2std, [
+            ('affine_transform', 'in3'),
+            ('displacement_field', 'in4'),
+        ]),
+        # Because std2anat are inverse transforms, warp is first
+        (dis_std2int, order_std2anat, [
+            ('affine_transform', 'in2'),
```

Comment:
An earlier function makes sure that the "affine_transform" is actually the mat file, right?

Reply:
Yeah, this should always just grab the AffineTransform

https://github.com/nipreps/nibabies/blob/d155295ae64e7b11d5047a5c44cb4839b87592fd/nibabies/interfaces/patches.py#L131-L144